src/ctia/entity/event/crud.clj

```diff
@@ -0,0 +1,74 @@
+(ns ctia.entity.event.crud
+  (:require
+   [schema.core :as s]
+   [ctia.entity.event.schemas
+    :refer [Event PartialEvent]]
+   [ctia.stores.es.crud :as crud]
+   [clj-momo.lib.es
+    [document :as document]
+    [schemas :refer [ESConnState SliceProperties]]
+    [slice :refer [get-slice-props]]]))
+
+(defn attach-bulk-fields [index event]
+  (assoc event
+         :_type "event"
+         :_id (:id event)
+         :_index index))
+
+(s/defn index-produce
+  "produce an event to an aliased index"
+  [state :- ESConnState
+   slice-props :- SliceProperties
+   events :- [Event]]
+  (document/bulk-create-doc
+   (:conn state)
+   events
+   (get-in state [:props :refresh] false)))
+
+(s/defn simple-produce
+  "produce an event to an index"
+  [state :- ESConnState
+   events :- [Event]]
+  (document/bulk-create-doc
+   (:conn state)
+   (->> events
+        (map (partial attach-bulk-fields (:index state))))
+   (get-in state [:props :refresh] false)))
+
+(s/defn produce
+  ([state :- ESConnState
+    slice-props :- SliceProperties
+    events :- [Event]]
+   (index-produce state slice-props events))
+
```

Comment:
_style remark_: I think this is an unnecessary newline, while I think it is tolerated by https://github.com/bbatsov/clojure-style-guide#optional-new-line-after-fn-name all example given don't use any empty lines. So I would get rid of them when possible.